File: backend/app/schemas/tradingview_signal.py
"""TradingViewSignal Pydantic スキーマ"""

from __future__ import annotations
from datetime import datetime
from typing import Optional, Any, Dict
from pydantic import BaseModel


class TradingViewSignalCreate(BaseModel):
    recommendation: Optional[str] = None
    score: Optional[float] = None
    buy_count: Optional[int] = None
    sell_count: Optional[int] = None
    neutral_count: Optional[int] = None
    ma_recommendation: Optional[str] = None
    osc_recommendation: Optional[str] = None
    details: Optional[Dict[str, Any]] = None


class TradingViewSignalResponse(BaseModel):
    id: int
    symbol: str
    recommendation: Optional[str] = None
    score: Optional[float] = None
    buy_count: Optional[int] = None
    sell_count: Optional[int] = None
    neutral_count: Optional[int] = None
    ma_recommendation: Optional[str] = None
    osc_recommendation: Optional[str] = None
    details: Optional[Dict[str, Any]] = None
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/tradingviewSignal.ts
export interface TradingViewSignal {
  id: number;
  symbol: string;
  recommendation: string | null;
  score: number | null;
  buy_count: number | null;
  sell_count: number | null;
  neutral_count: number | null;
  ma_recommendation: string | null;
  osc_recommendation: string | null;
  details: Record<string, unknown> | null;
  updated_at: string;
}

export interface TradingViewSignalCreate {
  recommendation?: string | null;
  score?: number | null;
  buy_count?: number | null;
  sell_count?: number | null;
  neutral_count?: number | null;
  ma_recommendation?: string | null;
  osc_recommendation?: string | null;
  details?: Record<string, unknown> | null;
}
